autoscheduler/scraper/tests/banner_requests_tests.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+
+import asyncio
+from aiohttp import ClientSession
+from scraper.banner_requests import generate_session_id, get_term_code
+from scraper.banner_requests import Semester, Location, BannerRequests
+from .aio_test_case import AioTestCase
+
+class GenerateSessionIDTests(unittest.TestCase):
+    """ Tests generate_session_id from banner_requests.py """
+    def test_is_18_chars(self):
+        """ Checks that the session_id is 18 characters long """
+
+        # Arrange/Act
+        session_id = generate_session_id()
+
+        # Assert
+        length = len(session_id)
+
+        assert length == 18 # Should be 18 characters long
+
+class GetTermCodeTests(unittest.TestCase):
+    """ Tests get_term_code from banner_requests.py """
+
+    def test_fall_college_station_returns_31(self):
+        """ Tests if when given fall semester & at college station if it returns 31 """
+        # Arrange
+        semester = Semester.fall
+        loc = Location.college_station
+
+        # Act
+        term_code = get_term_code("2019", semester, loc)
+
+        # Assert
+        assert term_code[4:] == "31" # term_code is 201931, but we just want last 2 chars
+
+    def test_spring_galveston_returns_22(self):
+        """ Tests if when given spring & at galveston, if the given term ends in 22 """
+        # Arrange
+        semester = Semester.spring
+        loc = Location.galveston
+
+        # Act
+        term_code = get_term_code("2019", semester, loc)
+
+        # Assert
+        assert term_code[4:] == "12"
+
+    def test_invalid_year_does_raise_error(self):
+        """ Provides an invalid year(where len != 4) and passes if it throws an error """
+        try:
+            _ = get_term_code("9", Semester.fall, Location.college_station)
+
+            assert False # If it doesn't raise an error, then didn't pass
+        except ValueError:
+            assert True
+            return
+
+class BannerRequestsTests(AioTestCase):
+    """ Tests BannerRequests for functionality. These access the network for communicating
```

Comment:
I'd like to have not only more tests, but for the current ones to be more descriptive/intentional in their functionality. The tests should really be examples for how they need to function, rather than basically testing that they don't throw errors.

For instance, testing that the first result of Fall 2019's departments isn't a good example of how the functions being used, as that fact isn't really that pertinent to the purpose of the function